Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean

```diff
@@ -645,6 +647,33 @@ end Point
 
 end WeierstrassCurve.Affine
 
+namespace WeierstrassCurve.Projective.Point
+
+/-! ## Weierstrass curves in projective coordinates -/
+
+variable {F : Type u} [Field F] {W : Projective F}
+
+lemma add_left_neg (P : W.Point) : -P + P = 0 :=
+  toAffine_addEquiv.injective <| by
+    rcases P with @⟨⟨_⟩, _⟩
+    simp only [map_add, toAffine_addEquiv_apply, toAffine_lift_neg, Affine.Point.add_left_neg,
+      toAffine_lift_zero]
+
+lemma add_comm (P Q : W.Point) : P + Q = Q + P :=
+  toAffine_addEquiv.injective <| by simp_rw [map_add, Affine.Point.add_comm]
+
+lemma add_assoc (P Q R : W.Point) : P + Q + R = P + (Q + R) :=
+  toAffine_addEquiv.injective <| by simp only [map_add, Affine.Point.add_assoc]
```

Comment:
inline

Reply:
I'll apply this to the downstream PR after the first one is merged.